model/common/src/icon4py/model/common/grid/grid_manager.py

```diff
@@ -26,495 +36,663 @@ def __init__(self, *args, **kwargs):
+    CONTROL_CELLS = "refin_c_ctrl"
 
-        #: start indices of horizontal grid zones for cell fields
-        START_INDEX_CELLS = "start_idx_c"
+    #: refine control value of edge indices
+    CONTROL_EDGES = "refin_e_ctrl"
 
-        #: start indices of horizontal grid zones for edge fields
-        START_INDEX_EDGES = "start_idx_e"
+    #: refine control value of vertex indices
+    CONTROL_VERTICES = "refin_v_ctrl"
 
-        #: start indices of horizontal grid zones for vertex fields
-        START_INDEX_VERTICES = "start_idx_v"
+    #: start indices of horizontal grid zones for cell fields
+    START_INDEX_CELLS = "start_idx_c"
 
-        #: end indices of horizontal grid zones for cell fields
-        END_INDEX_CELLS = "end_idx_c"
+    #: start indices of horizontal grid zones for edge fields
+    START_INDEX_EDGES = "start_idx_e"
 
-        #: end indices of horizontal grid zones for edge fields
-        END_INDEX_EDGES = "end_idx_e"
+    #: start indices of horizontal grid zones for vertex fields
+    START_INDEX_VERTICES = "start_idx_v"
 
-        #: end indices of horizontal grid zones for vertex fields
-        END_INDEX_VERTICES = "end_idx_v"
+    #: end indices of horizontal grid zones for cell fields
+    END_INDEX_CELLS = "end_idx_c"
 
-    def __init__(self, dataset: Dataset):
-        self._dataset = dataset
-        self._log = logging.getLogger(__name__)
+    #: end indices of horizontal grid zones for edge fields
+    END_INDEX_EDGES = "end_idx_e"
 
-    def dimension(self, name: GridFileName) -> int:
+    #: end indices of horizontal grid zones for vertex fields
+    END_INDEX_VERTICES = "end_idx_v"
+
+
+class GridFile:
+    """Represent and ICON netcdf grid file."""
+
+    INVALID_INDEX = -1
+
+    def __init__(self, file_name: str):
+        self._filename = file_name
+
+    def dimension(self, name: DimensionName) -> int:
+        """Read a dimension with name 'name' from the grid file."""
         return self._dataset.dimensions[name].size
 
-    def int_field(self, name: GridFileName, transpose=True, dtype=gtx.int32) -> np.ndarray:
-        try:
-            nc_variable = self._dataset.variables[name]
+    def attribute(self, name: PropertyName) -> Union[str, int, float]:
+        "Read a global attribute with name 'name' from the grid file."
+        return self._dataset.getncattr(name)
+
+    def int_variable(
+        self, name: FieldName, indices: xp.ndarray = None, transpose: bool = True
+    ) -> xp.ndarray:
+        """Read a integer field from the grid file.
+
+        Reads as int32.
+
+        Args:
+            name: name of the field to read
+            transpose: flag to indicate whether the file should be transposed (for 2d fields)
+        Returns:
+            xp.ndarray: field data
+
+        """
+        _log.debug(f"reading {name}: transposing = {transpose}")
+        data = self.variable(name, indices, dtype=gtx.int32)
+        return xp.transpose(data) if transpose else data
```

Comment:
Do you know whether the data is on the GPU device when read from netCDF? 
If the data is not on the device, I think we need to add `xp.asarray(xp.transpose(data)) if transpose else xp.asarray(data)` to return an object on the device. I suppose the connectivity needs to be on the device.